src/server/kernel/wtextinputv1.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2024 Yixue Wang <wangyixue@deepin.org>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+#pragma once
+
+#include <wglobal.h>
+#include <qwcompositor.h>
+
+#include <QObject>
+
+WAYLIB_SERVER_BEGIN_NAMESPACE
+struct ws_text_input_v1;
+struct ws_text_input_manager_v1;
+class WSurface;
+class WSeat;
+class WTextInputV1;
+class WTextInputManagerV1Private;
+class WTextInputV1Private;
+class WTextInputManagerV1 : public QObject, public WObject
+{
+    Q_OBJECT
+    W_DECLARE_PRIVATE(WTextInputManagerV1)
+public:
+    static WTextInputManagerV1 *create(QW_NAMESPACE::QWDisplay *display);
+
+Q_SIGNALS:
+    void newTextInput(WTextInputV1 *ti);
+
+private:
+    explicit WTextInputManagerV1(ws_text_input_manager_v1 *handle);
+};
+
+class WTextInputV1 : public QObject, public WObject
+{
+    Q_OBJECT
+    W_DECLARE_PRIVATE(WTextInputV1)
+    Q_PROPERTY(WSeat *seat READ seat NOTIFY seatChanged FINAL)
+    Q_PROPERTY(WSurface *focusedSurface READ focusedSurface NOTIFY focusedSurfaceChanged FINAL)
+    Q_PROPERTY(QString surroundingText READ surroundingText NOTIFY surroundingTextChanged FINAL)
+    Q_PROPERTY(uint surroundingCursor READ surroundingCursor NOTIFY surroundingTextChanged FINAL)
+    Q_PROPERTY(uint surroundingAnchor READ surroundingAnchor NOTIFY surroundingTextChanged FINAL)
+    Q_PROPERTY(uint contentHint READ contentHint NOTIFY contentTypeChanged FINAL)
+    Q_PROPERTY(uint contentPurpose READ contentPurpose NOTIFY contentTypeChanged FINAL)
+    Q_PROPERTY(QRect cursorRectangle READ cursorRectangle NOTIFY cursorRectangleChanged FINAL)
+    Q_PROPERTY(QString preferredLanguage READ preferredLanguage NOTIFY preferredLanguageChanged FINAL)
+
+public:
+    WSeat *seat() const;
+    WSurface *focusedSurface() const;
+    QString surroundingText() const;
+    uint surroundingCursor() const;
+    uint surroundingAnchor() const;
+    uint contentHint() const;
+    uint contentPurpose() const;
+    QRect cursorRectangle() const;
+    QString preferredLanguage() const;
+    wl_client *waylandClient() const;
```

Comment:
WObjectPrivate has been waylandClient function, so this public interface can add to WObject?

Reply:
I don't know if this interface is necessary in other protocol entities. But if export this in base class. That might be convenient.

Reply:
For my design, exporting waylandClient might add complexity, I will move all client's checks to private classes.

Reply:
Done.